packages/MapFactory/src/DTK_C_API_Map.cpp

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * Copyright (c) 2012-2018 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <DTK_C_API.h>
+#include "DTK_C_API_Map.hpp"
+
+#include <cerrno>
+#include <set>
+
+//---------------------------------------------------------------------------//
+namespace DataTransferKit
+{
+
+// We store the reinterpret_cast versions of pointers
+static std::set<void *> valid_map_handles;
+
+//---------------------------------------------------------------------------//
+
+} // end namespace DataTransferKit
+
+//---------------------------------------------------------------------------//
+
+extern "C" {
+
+//---------------------------------------------------------------------------//
+DTK_MapHandle DTK_createMap( DTK_ExecutionSpace space,
+                             MPI_Comm comm,
+                             DTK_UserApplicationHandle source,
+                             DTK_UserApplicationHandle target )
+{
+    if ( !DTK_is_initialized() )
+    {
+        errno = DTK_UNINITIALIZED;
+        return nullptr;
+    }
+
+    // For demonstration purposes just use the nearest neighbor map.
+    auto handle = reinterpret_cast<DTK_MapHandle>(
```

Comment:
I am not sure how I feel about this `reinterpret_cast`.
Should `createMap` return type be `DTK_MapHandle` if its only purpose is to support the C interface?

Reply:
Ping.

Reply:
This is needed to hide the C++ class pointer we actually use under the opaque handle type. Not sure you can do this with a static cast or any other type of cast.